Corzbank.Services/DepositService.cs

```diff
@@ -48,9 +51,33 @@ public async Task<DepositDTO> GetDepositById(Guid id)
             return result;
         }
 
+        public async Task<IEnumerable<DepositDTO>> GetDepositsForUser(Guid userId)
+        {
+            var deposits = await _depositRepo
+                .GetQueryable()
+                .Include(c => c.Card).ThenInclude(u => u.User)
+                .Where(u => u.Card.User.Id == userId)
+                .ToListAsync();
+
+            var mappedDeposits = _mapper.Map<IEnumerable<DepositDTO>>(deposits);
+
+            return mappedDeposits;
+        }
+
         public async Task<DepositDTO> OpenDeposit(DepositModel deposit)
         {
+            var cardForDeposit = await _cardRepo.GetQueryable().FirstOrDefaultAsync(c => c.Id == deposit.CardId);
+
+            if (cardForDeposit.Balance < deposit.Amount)
+                return null;
```

Comment:
You should throw the exception in such case, returning null means nothing for someone who made a request. Create a custom exceptions, like (NotFoundException, ForbiddenException). In our case, you should throw ForbiddenException

Example how to create an exception:
public class ForbiddenException : RequestException
    {
        public override int StatusCode => 403;
    }
    public class RequestException : ApplicationException
    {
        public RequestException() { }
        public RequestException(string message)
            : base(message)
        {

        }
        public virtual int StatusCode => 403;
    }

Reply:
resolved